app/src/main/java/org/jellyfin/mobile/MainActivity.kt

```diff
@@ -125,18 +128,26 @@ class MainActivity : AppCompatActivity(), WebViewController {
                 val url = request.url
                 val path = url.path?.toLowerCase(Locale.ROOT) ?: return null
                 return when {
-                    path.endsWith(Constants.INDEX_PATH) -> {
-                        val patchedIndex = loadPatchedIndex(httpClient, url.toString())
-                        if (patchedIndex != null) {
-                            runOnUiThread { connectionHelper.onConnectedToWebapp() }
-                            patchedIndex
-                        } else {
-                            runOnUiThread { connectionHelper.onErrorReceived() }
-                            emptyResponse
+                    path.endsWith("scripts/apploader.js") -> {
+                        runOnUiThread {
+                            webView.evaluateJavascript(
```

Comment:
Can we load the JavaScript from assets instead of having it hardcoded in the Kotlin code?

Reply:
Should be doable, but loading it async might increase the possibility of race-conditions. Maybe I can move it to a constant?

Reply:
Hmm, I don't think it matters too much. Apploader should load first anyway because it contains the `require` function etc.

Reply:
I believe the first check for NativeInterface is in `site.js` though, iirc. We should be loaded before that.